### PR TITLE
Add support for /schema /schema{identifier} in the redfish API

### PR DESCRIPTION
**Pull request by tannoa2**
@jlongever @zyoung51 @amymullins @keedya @uppalk1

### CODE REVIEW

**Comment by tannoa2** [comment]
Test this please

**Comment by tannoa2** [comment]
test this please

**Comment by tannoa2** [comment]
Test this please

File: lib/api/redfish-1.0/schemas.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+var listSchemas = controller(function(req, res) {
+    var options = redfish.makeOptions(req, res);
+    return Promise.props({
+        schemasObj: redfish.getSchemas()
+    }).then(function(data){
+        _.forEach(data.schemasObj, function(value, idx){
+            data.schemasObj[idx] = (value.split('/v1/'))[1];
+            //removing the .json extension
+            data.schemasObj[idx] = (data.schemasObj[idx].split('.jso'))[0];
```
Comment (yyscamper) [inline lib/api/redfish-1.0/schemas.js]: `.json`?

Comment (tannoa2) [inline lib/api/redfish-1.0/schemas.js]: HI Felix, there is no need to spell out json for the operation to work as intended

Comment (yyscamper) [inline lib/api/redfish-1.0/schemas.js]: OK, just a reminder you didn't have any typo. 
Actually, your two splitting operation can be merged into a single one, I am OK with either way:
```javascript
> "/redfish/v1/myschema.json.1.0.0".split(/\/v1\/(\w+).json/)[1]
'myschema'
```

**Comment by yyscamper** [comment]
:+1:

**Comment by keedya** [comment]
:+1:

File: lib/api/redfish-1.0/schemas.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    });
+});
+
+var getSchema = controller(function(req, res){
+    var identifier = req.swagger.params.identifier.value;
+    var options = redfish.makeOptions(req, res, identifier);
+    options.identifier = identifier;
+    return redfish.render('redfish.1.0.0.jsonschemafile.1.0.0.json',
+        'JsonSchemaFile.1.0.0.json#/definitions/JsonSchemaFile',
+        options);
```
Comment (zyoung51) [inline lib/api/redfish-1.0/schemas.js]: Can you add the .catch handling here?

**Comment by tannoa2** [comment]
test this please

**Comment by tannoa2** [comment]
test this please

**Comment by tannoa2** [comment]
test this please

**Pull request merged**